pyo3-macros-backend/src/params.rs

```diff
@@ -310,12 +310,18 @@ fn impl_arg_param(
 
         Ok(quote_arg_span! {
             let #mut_ _tmp: #target_ty = #arg_value_or_default?;
-            #[allow(clippy::needless_option_as_deref)]
+            #[allow(clippy::needless_option_as_deref, clippy::borrow_deref_ref)]
             let #arg_name = #borrow_tmp;
         })
-    } else {
+    } else if cfg!(feature = "pyproto") {
```

Comment:
Lack of `remove_lifetime` is probably why `pyproto` broke - it inserts a hidden lifetime to try and behave like lifetime elision :(. Super weird and so close to killing off now!